pkg/cmd/server/storage/server.go

```diff
@@ -76,6 +79,102 @@ func (s *StorageRESTServer) Run(stopCh <-chan struct{}) {
 	close(ch)
 }
 
+func (s *StorageRESTServer) sync(key string) (*types.StorageBody, error) {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+	sb, ok := s.pushedLocations[key]
+	if ok {
+		return sb, nil
+	}
+	ssb, err := s.st.Fetch(key)
+	return &ssb, err
+}
+
+func (s *StorageRESTServer) del(key string) {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+	delete(s.pushedLocations, key)
+}
+
+func (s *StorageRESTServer) handleCatalogCurrentKeySetPost(c *gin.Context) {
+	key := c.Query("key")
+	// no content for the key QP means no models were discovered
+
+	keys := strings.Split(key, ",")
+	keyHash := map[string]struct{}{}
+	if len(key) > 0 {
+		for _, k := range keys {
+			keyHash[k] = struct{}{}
+		}
+	}
+
+	var err error
+	currentKeys := []string{}
+	currentKeys, err = s.st.List()
+	if err != nil {
+		c.Status(http.StatusInternalServerError)
+		msg := fmt.Sprintf("error listing location keys: %s", err.Error())
+		klog.Errorf(msg)
+		c.Error(fmt.Errorf(msg))
+		return
+	}
+
+	var errors []error
+	for _, k := range currentKeys {
+		_, ok := keyHash[k]
+		if !ok {
+			msg := ""
+			//TODO for summit we were not going to"aggressively" inform backstage of deletions by leveraging
+			// the delete location catalog REST API; however, with location testing
+			// https://github.com/redhat-ai-dev/rhdh-plugins/blob/6b0c4a21c1cdfeba4cf2618d4aabadff544c7efc/workspaces/rhdh-ai/plugins/catalog-backend-module-rhdh-ai/src/providers/RHDHRHOAIEntityProvider.ts#L198-L202
```

Comment:
Is this comment still valid with your latest changes from https://github.com/redhat-ai-dev/rhdh-plugins/pull/4?

Reply:
yes it is still valid .... the EntityProvider was not even successfully deleting the location it tried to delete, much less the entities brought in by the location

https://github.com/redhat-ai-dev/rhdh-plugins/pull/4 fixes it so that **IF** we call `s.bkstg.DeleteLocation(...)` we no delete not just the location, but the entities brought in by the location.